cdap-app-templates/cdap-etl/cdap-data-streams-base/src/main/java/io/cdap/cdap/datastreams/DataStreamsSparkLauncher.java

```diff
@@ -97,7 +97,13 @@ public void initialize() throws Exception {
     }
 
     SparkConf sparkConf = new SparkConf();
+    // required spark configs to prevent data loss during real-time pipeline upgrades
     sparkConf.set("spark.streaming.backpressure.enabled", "true");
+    sparkConf.set("spark.streaming.receiver.writeAheadLog.enable", "true");
+    sparkConf.set("spark.streaming.receiver.writeAheadLog.closeFileAfterWrite", "true");
+    sparkConf.set("spark.streaming.driver.writeAheadLog.closeFileAfterWrite", "true");
+    sparkConf.set("spark.streaming.receiver.writeAheadLog.rollingIntervalSecs", String.valueOf(0));
```

Comment:
what does this mean? It doesn't roll over?

Reply:
This is mainly to work better with GCS - https://groups.google.com/g/cdap-issues/c/jbO6i0uKKKk/m/fCPzp4kzCgAJ

Reply:
I see, can you add a comment in the code explaining